Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.h

```diff
@@ -67,14 +68,14 @@ class SharedVideoFrameReader;
 class LibWebRTCCodecsProxy final : public IPC::WorkQueueMessageReceiver {
     WTF_MAKE_TZONE_ALLOCATED(LibWebRTCCodecsProxy);
 public:
-    static Ref<LibWebRTCCodecsProxy> create(GPUConnectionToWebProcess&);
+    static Ref<LibWebRTCCodecsProxy> create(GPUConnectionToWebProcess&, SharedPreferencesForWebProcess&);
     ~LibWebRTCCodecsProxy();
     void stopListeningForIPC(Ref<LibWebRTCCodecsProxy>&& refFromConnection);
     bool allowsExitUnderMemoryPressure() const;
-
-private:
+    const SharedPreferencesForWebProcess& sharedPreferencesForWebProcess() const { return m_sharedPreferencesForWebProcess; }
+    void updateSharedPreferencesForWebProcess(SharedPreferencesForWebProcess sharedPreferencesForWebProcess) { m_sharedPreferencesForWebProcess = WTFMove(sharedPreferencesForWebProcess); }
```

Comment:
Since we are writing in main thread and reading in queue, there is an inherent race.
I guess this is fine since for the security case, the parameters should be stable (or at least start very strict).

Reply:
I think it's more safe to make `m_sharedPreferencesForWebProcess` only accessed from `WorkQueue`, by dispatching task to `m_queue` for initialization and update.